src/main/java/io/confluent/connect/elasticsearch/Validator.java

```diff
@@ -117,6 +125,42 @@ private void validateCredentials() {
     }
   }
 
+  private void validateDataStreamFields() {
+    if (config.dataStreamType() == DataStreamType.NONE ^ config.dataStreamDataset().equals("")) {
+      String errorMessage = String.format(
+          "Either both or neither '%s' and '%s' must be set.",
+          DATA_STREAM_DATASET_CONFIG,
+          DATA_STREAM_TYPE_CONFIG
+      );
+      addErrorMessage(DATA_STREAM_TYPE_CONFIG, errorMessage);
+      addErrorMessage(DATA_STREAM_DATASET_CONFIG, errorMessage);
+    }
+    boolean atLeastOneSet = config.dataStreamType() != DataStreamType.NONE
+        || !config.dataStreamDataset().equals("");
```

Comment:
i suspect that you will need  a method `boolean useDataStream/isDataStream()` later so you can add one now to the config class and use it here